modules/nlu/src/views/entities/editor.jsx

```diff
@@ -0,0 +1,168 @@
+
+      this.setState({ currentEntity: newEntity, currentOccurence: newEntity && newEntity.occurences[0] })
+    }
+  }
+
+  onEnterPressed = (event, cb) => {
+    const enterKey = 13
+    if (event.keyCode === enterKey) {
+      cb()
+    }
+  }
+
+  addSynonym = (occurence, synonym) => {
+    let entity = this.state.currentEntity
+
+    if (occurence.synonyms.includes(synonym)) {
+      return
+    }
+
+    occurence.synonyms = [...occurence.synonyms, synonym]
+    const index = entity.occurences.findIndex(o => o.name === occurence.name)
+    entity.occurences[index] = occurence
+
+    this.setState({ currentEntity: entity }, this.onUpdate)
+  }
+
+  removeSynonym = (occurence, index) => {
+    let entity = this.state.currentEntity
+
+    occurence.synonyms.splice(index, 1)
+
+    const eIndex = entity.occurences.findIndex(o => o.name === occurence.name)
+    entity.occurences[eIndex] = occurence
+    this.setState({ currentEntity: entity }, this.onUpdate)
+  }
+
+  onSynonymInputChange = event => {
+    const value = event.target.value
+    this.setState({ synonymInput: value })
+  }
+
+  addOccurence = () => {
+    const occurence = this.state.occurenceInput
+    let entity = this.state.currentEntity
+
+    if (entity.occurences.includes(occurence)) {
+      return
+    }
+
+    entity.occurences = [...entity.occurences, { name: occurence, synonyms: [] }]
+
+    this.occurenceInputRef.current.value = ''
+    this.setState({ currentEntity: entity, occurenceInput: '' }, this.onUpdate)
+  }
+
+  handleOccurenceEnter = event => {
+    this.onEnterPressed(event, this.addOccurence)
+  }
+
+  onOccurenceInputChange = event => {
+    const value = event.target.value
+    this.setState({ occurenceInput: value })
+  }
+
+  selectOccurence = occurence => {
+    if (occurence !== this.state.currentOccurence) {
+      this.setState({ currentOccurence: occurence })
+    }
+  }
+
+  removeOccurence = occurence => {
+    const confirm = window.confirm(
+      `Are you sure you want to delete the occurence "${occurence.name}" and all its synonyms?`
+    )
+    if (!confirm) {
+      return
+    }
+
+    const entity = this.state.currentEntity
+    const index = entity.occurences.findIndex(o => o.name === occurence.name)
+
+    entity.occurences.splice(index, 1)
+    this.setState({ currentEntity: entity, currentOccurence: entity.occurences[0] }, this.onUpdate)
+  }
+
+  renderOccurences = () => {
+    const occurences = this.state.currentEntity && this.state.currentEntity.occurences
+    if (!occurences) {
+      return null
+    }
+
+    return (
+      <div>
+        <input
+          class={classNames('form-control', style.occurenceInput)}
+          ref={this.occurenceInputRef}
+          type="text"
+          placeholder="Type to create an occurence"
+          onKeyDown={this.handleOccurenceEnter}
+          onChange={this.onOccurenceInputChange}
+        />
+        {occurences.map(o => (
+          <ListGroupItem className={style.occurence}>
```

Comment:
You should give it a key, e.g.. `key={'occurance-' + o.name}` or something similar
https://reactjs.org/docs/lists-and-keys.html